### ISSUE TITLE
Bad location caching for pubspec lint reporting.

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#40]
Specifically, this
```
library library_names;
```
yields:

```
[lint] DO use `lowercase_with_underscores` for package names. (tool/rule.dart, line 1, col 6)
```

(And shouldn't.)

**Comment by pq** [comment]
Actually, I think this is a bad location caching issue.  Specifically:
```
[lint] AVOID wrapping fields in getters and setters just to be "safe". (test/rules/unnecessary_getters_setters.dart, line 18, col 3)
[lint] AVOID wrapping fields in getters and setters just to be "safe". (test/rules/unnecessary_getters_setters.dart, line 7, col 3)
[lint] DO use `lowercase_with_underscores` for package names. (tool/rule.dart, line 1, col 6)
```

I think the last entry should be reporting the pubspec location.
